src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
```

Comment:
So, we can do the same by reduce
But, will be better if we create one more function for this logic, bc we repeat yourself

So, create the function calculateTotalAges which takes filteredPeople & returns totalAges

```suggestion
  const totalAges = getTotalAges(filteredByFemaleGender);
```